trace/trace.go

```diff
@@ -244,8 +243,23 @@ func SpanFromContext(c context.Context) *Trace {
 	return StartChildSpan(parent)
 }
 
-// StartSpanFromContext is used to create a child span
-// when the parent trace is in the context
+// StartSpanFromContext creates a span that fits into a span tree -
+// child or root. It makes a new span, find a parent span if it exists
+// on the context and fills in the data in the newly-created span to
+// ensure a proper parent-child relationship (if no parent exists, is
+// sets the new span up as a root Span, i.e., a Trace). Then
+// StartSpanFromContext sets the newly-created span as the current
+// span on a child Context. That child context is returned also.
+//
+// Name inference
+//
+// StartSpanFromContext takes a name argument. If passed "", it will
+// infer the current function's name as the span name.
+//
+// Stability
```

Comment:
small nit: I don't love marking "stability" header for the function which is recommended and considered stable (as opposed to the one which isn't). I don't think we need to draw attention to it in the same way, so "...name as the span name. StartSpanFromContext is the recommended way to create trace spans in a program." would work just as well.

Reply:
Yeah agreed. I was looking for something that conveys "we really want you to use this over that other thing"; will iterate.